source/base/data_out_base.cc

```diff
@@ -7544,6 +7544,80 @@ DataOutInterface<dim, spacedim>::write_pvtu_record(
 }
 
 
+template <int dim, int spacedim>
+void
+DataOutInterface<dim, spacedim>::write_vtu_with_pvtu_record(
+  const std::string &folder,
+  const std::string &filename_without_extension,
+  const unsigned int counter,
+  const unsigned int n_digits_counter,
+  const MPI_Comm &   mpi_communicator,
+  const unsigned int n_groups) const
+{
+  const unsigned int rank = Utilities::MPI::this_mpi_process(mpi_communicator);
+  const unsigned int n_ranks =
+    Utilities::MPI::n_mpi_processes(mpi_communicator);
+  const unsigned int n_files_written =
+    (n_groups == 0 || n_groups > n_ranks) ? n_ranks : n_groups;
+
+  const unsigned int n_digits =
+    std::ceil(std::log10(std::fabs(n_files_written)));
+
+  const unsigned int color =
```

Comment:
can't you use ``rank % n_files_written``?

Reply:
yes, it should be the same

Reply:
done in https://github.com/dealii/dealii/pull/8904/commits/f07a8bf9bce24520305b079bc179599d78782753